awswrangler/distributed/datasources/pandas_text_datasource.py

```diff
@@ -58,35 +77,128 @@ def _read_stream(  # type: ignore  # pylint: disable=arguments-differ
     def _read_file(self, f: pyarrow.NativeFile, path: str, **reader_args: Any) -> pd.DataFrame:
         raise NotImplementedError()
 
-    def _write_block(  # type: ignore  # pylint: disable=signature-differs
+    def do_write(  # type: ignore  # pylint: disable=arguments-differ
```

Comment:
Is there any difference between this and parquet dataset? Shall we create a parent class?

Reply:
Yes. The Parquet dataset is opening the file using PyArrow's `FileSystem`. Here, I am using our own `open_s3_object` function. Another consequence of this is that in the Parquet data source, the `s3://` prefix gets removed, whereas here I am keeping it.

Reply:
Is it actually using PyArrow system in the runtime or is it just what the typing says? Because we should be using S3 for Parquet dataset, too

Anyways, I think there's potential to refactor but we can do it later 👍